pageserver/src/layered_repository.rs

```diff
@@ -1523,6 +1538,15 @@ impl LayeredTimeline {
                 &metadata,
                 false,
             )?;
+
+            // Update upload metrics
+            for path in vec![new_delta_path.clone()] {
```

Comment:
This all should go below `self.upload_layers.load(atomic::Ordering::Relaxed)`, the upload is not enabled by default and not happening.

Also, the whole `vec![new_delta_path.clone()]` seems to be redundant, since `File` is capable of opening `&Path` too?

Reply:
Ah sorry the vec! stuff slipped. I'll clean it up.

I want a way to count without actually uploading, which is great for tests. I guess there are better ways to write that code to make sure we don't forget to count. I'll see if there's an easy fix.

Reply:
Another way to achieve this would be adding a tag to actually display, if this data was sent to the remote storage and add more logic to determine that tag, something like: 

```
static ref BYTES_SCHEDULED_FOR_UPLOAD: zenith_metrics::IntCounterVec =
    zenith_metrics::register_int_counter_vec!(
        "pageserver_bytes_scheduled_for_upload",
        "Total bytes estimated to be uploaded to the cloud storage, tagged by an actual upload event [uploaded|skipped]",
        &["uploaded", "skipped"]
    )
    .expect("failed to define a bytes scheduled for upload metric");
```

Reply:
I dialed back the language so that the names don't suggest files were actually uploaded. I want to avoid adding test-only metrics to prod.